features/my_google.feature

```diff
@@ -2,7 +2,7 @@
 Feature: As a web automation tester in training I want to automate basic searching with Google
 
   #A scenario is a single test, a collection of Given, When, and Then steps
-  Scenario: Basic search
+  Scenario Outline: Basic search
```

Comment:
So a Scenario Outline that iterates one or more time based on parameters in a table is one way to have a step that works for multiple pages. However, since your step definition looks like this:
```ruby
Then(/^the (.*) Page has Kittens$/) do |page_name|
  ...
end
```

You should just be able to have this be a regular Scenario that looks like:
```ruby
Scenario: Basic Search
  Given I am on the Google Home Page
  When I search for Kittens
  Then I see the Google Search Results Page
  And the search_results Page has Kittens
  ....
```

You could even use "And the Search Results Page has Kittens" and have a to_method method somewhere that just translates Search Results to search_results.

```ruby
def to_method(str)
  str.downcase.gsub(' ', '_')
end
```